packages/template-typescript-minimal/app/pages/home.tsx

```diff
@@ -101,6 +103,12 @@ const Home = ({value}: Props) => {
             })
     )
 
+    const {res, isServerSide} = useServerContext()
+    if (isServerSide && query.data) {
+        console.log('--- useServerContext')
+        res.status(404)
+    }
```

Comment:
What it looks like now to call `useServerContext`.

Reply:
> I was under the impression that the callback was something we were in alignment on

@bendvc Yes, we were alignment in using callback. I remember what made you agree with using it is that the callback abstracts you from knowing when exactly is the right time to use `req` or `res`. 

But we didn't agree on the naming of the hook. Sorry I should've mentioned it in the PR description.. that's my main reason for not using the callback this time around. If there's a good name for it, I'll switch the implementation to use callback. But if there isn't a good name for it, I prefer a more familiar hook interface that returns an object instead. 

I think returning `isServerSide` is a good compromise. It still gives the users a way to know when to run their code. And they don't need to know about implementation detail of "is it prepass or not".

Reply:
👍